### PR TITLE
feat: add context token display to statusline command

### LINKED ISSUE #454: [Feature Request] Display current context token count in ccusage statusline
I would like to request a feature to display the current context token count in ccusage, similar to the implementation shown in this gist:
https://gist.github.com/9wick/8c0e9dae6c191cf3d7182315a1f1a81a

When working with LLMs, it's crucial to monitor the current context size to:
- Understand how much of the context window is being used
- Optimize prompts and responses to stay within limits
- Make informed decisions about when to summarize or truncate conversations
- Prevent unexpected autocompact triggers and maintain control over context management

### PR DESCRIPTION
**Pull request by ItsLucas**
Add context window usage tracking to the statusline hook output by:
- Reading transcript file provided by Claude Code statusline hook
- Extracting current input token count from transcript usage data
- Calculating percentage of 200K context window used
- Displaying tokens with thousand separators and color-coded percentage
  - Green: <50% (normal usage)
  - Yellow: 50-80% (moderate usage)
  - Red: >80% (high usage, approaching limit)

Format: 🧠 25,000 (13%) appears at end of statusline

Addresses user request for context window monitoring to help optimize prompt length and prevent unexpected context management issues.

Fixes #454 

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

- **New Features**
  - Status line now shows Context Usage (🧠 token count + colored percentage); defaults and thresholds are configurable.
- **Robustness**
  - If context data can’t be determined or errors occur, status line generation continues unaffected.
- **Documentation**
  - Guide updated with Context Usage explanation, examples for configuring thresholds, and validation behavior.
- **Tests**
  - Added tests covering transcript parsing, context calculation, threshold parsing, clamping, and edge cases.

### CODE REVIEW

**Comment by ryoppippi** [comment]
@ItsLucas good but can you move logic to data-loader.ts and import it from statusline.ts

**Comment by ItsLucas** [comment]
image
Seems the line is a little longer when the "approach limit" warning appears

File: src/data-loader.ts [hunk 1]
```diff
@@ -1335,6 +1336,115 @@ export async function loadBucketUsageData(
...
+		}
+	}
+
+	// No valid usage information found
+	logger.debug('No usage information found in transcript');
+	return null;
+}
+
+// Test for calculateContextTokens
+if (import.meta.vitest != null) {
```
Comment (ryoppippi) [inline src/data-loader.ts]: add test at the end

File: src/data-loader.ts [hunk 2]
```diff
@@ -1335,6 +1336,115 @@ export async function loadBucketUsageData(
...
+		}
+
+		try {
+			const obj = JSON.parse(trimmedLine) as TranscriptMessage;
+
+			// Check if this line contains the required token usage fields
+			if (obj.type === 'assistant'
+				&& obj.message != null
+				&& obj.message.usage != null
+				&& obj.message.usage.input_tokens != null) {
```
Comment (ryoppippi) [inline src/data-loader.ts:1447]: define zod schema, or extend the existing schema

File: src/_consts.ts [hunk 3]
```diff
@@ -140,6 +140,21 @@ export const BURN_RATE_THRESHOLDS = {
...
+/**
+ * Default Claude 4 context window limit (200,000 tokens)
+ * Used for calculating context usage percentages in statusline
+ */
+export const CONTEXT_LIMIT = 200_000;
+
+/**
+ * Context usage percentage thresholds for color coding
+ */
+export const CONTEXT_USAGE_THRESHOLDS = {
```
Comment (ryoppippi) [inline src/_consts.ts]: nice
but if we can define it as env vareiable it is really good

**Review by ryoppippi — APPROVED** [review]
Hi added some comments!
@ItsLucas 
PTAL
